File: src/utils/fileHelper.js
const fs = require('fs/promises');
const path = require('path');

async function saveResults(dateStr, results) {
  const dir = path.resolve(__dirname, '../../results');
  await fs.mkdir(dir, { recursive: true });
  const filePath = path.join(dir, `${dateStr}_signals.json`);
  await fs.writeFile(filePath, JSON.stringify(results, null, 2), 'utf-8');
  return filePath;
}

module.exports = { saveResults };

File: src/utils/file_helper.py
import json
from pathlib import Path


RESULTS_DIR = Path(__file__).resolve().parents[2] / 'results'


def save_results(date_str: str, results):
    RESULTS_DIR.mkdir(parents=True, exist_ok=True)
    file_path = RESULTS_DIR / f"{date_str}_signals.json"
    file_path.write_text(json.dumps(results, indent=2), encoding='utf-8')
    return file_path
